server/src/controllers/Events/resolver.ts

```diff
@@ -138,34 +160,57 @@ const hasVenueChanged = (data: EventInputs, event: EventWithUsers) =>
   data.venue_type !== event.venue_type ||
   (isOnline(event.venue_type) && data.streaming_url !== event.streaming_url) ||
   (isPhysical(event.venue_type) && data.venue_id !== event.venue_id);
+const hasDateChanged = (data: EventInputs, event: EventWithUsers) =>
+  !isEqual(data.ends_at, event.ends_at) ||
+  !isEqual(data.start_at, event.start_at);
 
 const buildEmailForUpdatedEvent = async (
   data: EventInputs,
   event: EventWithUsers,
 ) => {
-  const subject = `Venue changed for event ${event.name}`;
-  let venueDetails = '';
-
-  if (isPhysical(event.venue_type)) {
-    const venue = await prisma.venues.findUniqueOrThrow({
-      where: { id: data.venue_id },
-    });
-    // TODO: include a link back to the venue page
-    venueDetails += `The event is now being held at <br>
-${venue.name} <br>
-${venue.street_address ? venue.street_address + '<br>' : ''}
-${venue.city} <br>
-${venue.region} <br>
-${venue.postal_code} <br>
-`;
-  }
+  const subject = `Details changed for event ${event.name}`;
+  let venueContent: Array<string> = [];
 
-  if (isOnline(event.venue_type)) {
-    venueDetails += `Streaming URL: ${data.streaming_url}<br>`;
-  }
   // TODO: include a link back to the venue page
-  const body = `We have had to change the location of ${event.name}.<br>
-${venueDetails}`;
+  const createVenueContent = async () => {
+    const venueDetails = [];
+    let venue;
+    isPhysical(event.venue_type) &&
+      (venue = await prisma.venues.findUniqueOrThrow({
+        where: { id: data.venue_id },
+      }));
+    venueDetails.push(`The event is now being held at <br>
+${venue?.name} <br>
+${venue?.street_address && venue?.street_address + '<br>'}
+${venue?.city} <br>
+${venue?.region} <br>
+${venue?.postal_code} <br>
+`);
+    isOnline(event.venue_type) &&
+      venueDetails.push(`Streaming URL: ${data.streaming_url}<br>`);
+
+    return (venueContent = venueDetails);
+  };
+  {
+    hasVenueChanged(data, event) && createVenueContent();
+  }
+  let dateChangeContent = '';
+  hasDateChanged(data, event) &&
+    (dateChangeContent = `Updated dates <br />
+  ----------------------------<br />
+  ----------------------------<br />
+  <br />
+  Start: ${formatDate(data.start_at)}<br />
+  End: ${formatDate(data.ends_at)}<br />
+  ----------------------------<br />
+  <br />
+  `);
+  const body = `Updated venue details<br/>
+${venueContent?.join('')}<br />
+----------------------------<br />
+<br />
+${dateChangeContent}
+`;
```

Comment:
This is quite a confusing sequence.  All that really needs to happen is something like this

```ts
  const streamingUrlUpdates = hasStreamingUrlChanged(data, event)
    ? createStreamingUrlUpdates()
    : '';
  const dateUpdates = hasDateChanged(data, event)
    ? createDateUpdates()
    : '';
// ...
  return `a template string with the various updates inserted`
 ```

For example: 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/tools/challenge-helper-scripts/helpers/get-step-template.ts#L62-L89